### ISSUE TITLE
directory depth error

### ISSUE THREAD

**Issue by bendichter** [dandi/dandiarchive#728]
![image](https://user-images.githubusercontent.com/844306/120861711-b9800900-c555-11eb-829d-491dbbe5b4b6.png)

dandiset 108 is at least 9 directories deep. When I click on the 9th "0", the website teleports me back to level 1. I am not sure if this is a level issue or maybe there are too many elements to display or some other reason.

![image](https://user-images.githubusercontent.com/844306/120861946-1d0a3680-c556-11eb-9204-eb3a99e6d72d.png)

**Comment by satra** [comment]
that's a bad dataset (i stopped him on partial upload). he will fix it. i should really delete the files.

**Comment by waxlamp** [comment]
I suppose the remaining question is, how did 108 get to be in a bad state without being stopped by validation or some other serverside mechanism?

@satra, if we can get an understanding thereof, and @bendichter, if you're able to fix 108 to your satisfaction, then I think we can close this issue.

**Comment by satra** [comment]
there are some datasets that don't go through validation. these are datasets where we are working with the community. this was an example of someone directly uploading a zarr folder. and i had to stop early because there was a mismatch of concepts. our current backend would destroy the zarr structure, so it doesn't make sense to upload it.

**Comment by waxlamp** [comment]
Ah I see. So if these datasets are currently involving personalized care from a member of our team, I think we can close this issue.

**Comment by yarikoptic** [comment]
I don't think that we should dismiss this issue: even though it is a "non conventional" dataset, web UI must not just crash while traversing it.  We IMHO should take this opportunity as a lucky occasion that we managed to run into some unguarded assumption within the codebase and address it.  FWIW -- the jumping back to the top I still observe while going from https://gui.dandiarchive.org/#/dandiset/000108/draft/files?location=sub-MITU01%2Fses-20210119h14m35s15%2Fmicroscopy%2Fsub-MITU01_run-1_sample-mysample_stain-mystain_chunk-1_spim.ngff%2F0%2F0%2F0%2F0%2F0%2F to yet another `0/` but I do not see error message in console.

**Comment by yarikoptic** [comment]
The underlying issue seems to be within `dandi-api` though since 
```
$> curl -X GET "https://api.dandiarchive.org/api/dandisets/000108/versions/draft/assets/paths/?path_prefix=sub-MITU01/ses-20210119h14m35s15/microscopy/sub-MITU01_run-1_sample-mysample_stain-mystain_chunk-1_spim.ngff/0/0/0/0/0/" -H "accept: application/json"
["0/"]%                                                                                                                                                                                                              
```

but going under returns empty list:
```
$> curl -X GET "https://api.dandiarchive.org/api/dandisets/000108/versions/draft/assets/paths/?path_prefix=sub-MITU01/ses-20210119h14m35s15/microscopy/sub-MITU01_run-1_sample-mysample_stain-mystain_chunk-1_spim.ngff/0/0/0/0/0/0/" -H "accept: application/json"
[]%                 
```

which IIRC from https://github.com/dandi/dandi-api/pull/312 should not happen since we simply should have not had any of thos `0/` returned if there is no asset with the path under there.  But also web-ui should have not just jumped up if it encountered a "folder" without assets:  that should not happen but also IMHO it should have just shown an empty "folder" .

**Comment by waxlamp** [comment]
@yarikoptic is there a related issue filed in dandi-api?

**Comment by yarikoptic** [comment]
I don't know.  if it was, it would have probably been back-referenced here. So most likely - not, or was not yet searched-up/cross-referenced.
